File: Data Structures & Algorithms/missing-number/submission-1.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int xor_all = 0; 
        int xor_nums = 0; 

        for (int i = 0; i < nums.size()+1; ++i) { 
            xor_all ^= i;
        }
        
        for (int num : nums) { 
            xor_nums ^= num; 
        }

        return xor_all ^ xor_nums; 
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
from collections import defaultdict
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counter = defaultdict(int)

        for num in nums:
            counter[num] += 1
        
        sorted_counter = sorted(counter.items(), key=lambda pair : pair[1])
        return [list(sorted_counter)[-i][0] for i in range(1, k+1)]
